File: articles/future-generations-and-long-term-responsibility/c/inherited_burden_score.c
#include <stdio.h>

double inherited_burden(double climate, double debt, double infrastructure, double ecology, double institution, double tech_lock, double adaptive, double representation) {
    return 0.18 * climate + 0.14 * debt + 0.16 * infrastructure + 0.18 * ecology + 0.14 * tech_lock
        + 0.10 * (1.0 - institution) + 0.06 * (1.0 - adaptive) + 0.04 * (1.0 - representation);
}

int main(void) {
    printf("Short-Term Extraction inherited_burden: %.4f\n",
           inherited_burden(0.88, 0.76, 0.70, 0.84, 0.38, 0.62, 0.36, 0.24));
    return 0;
}

File: articles/future-generations-and-long-term-responsibility/julia/inherited_burden_score.jl
# Julia standard-library example for inherited burden scoring.

profiles = Dict(
    "Stewardship State" => [0.34, 0.32, 0.28, 0.30, 0.82, 0.36, 0.84, 0.80],
    "Short-Term Extraction" => [0.88, 0.76, 0.70, 0.84, 0.38, 0.62, 0.36, 0.24],
    "Reparative Intergenerational Compact" => [0.30, 0.36, 0.32, 0.26, 0.84, 0.34, 0.86, 0.86]
)

function inherited_burden(v)
    # v = climate debt infrastructure ecology institution tech_lock adaptive representation
    0.18*v[1] + 0.14*v[2] + 0.16*v[3] + 0.18*v[4] + 0.14*v[6] +
    0.10*(1-v[5]) + 0.06*(1-v[7]) + 0.04*(1-v[8])
end

println("scenario,inherited_burden")
for (name, values) in sort(collect(profiles))
    println("$(name),$(round(inherited_burden(values), digits=4))")
end
